hypothesis/lms
# Move email digests template into Python/Jinja2

Fixes https://github.com/hypothesis/lms/issues/5655.

Depends on https://github.com/hypothesis/lms/pull/5657/.

This PR moves the email digests template from Mailchimp into Python/Jinja2. This involves:

1. Creating a new Jinja2 template for the email at `lms/templates/email/instructor_email_digest/` based on the template that we previously had in Mailchimp. This is basically exactly the same as the template that we have in Mailchimp except that I fixed the formatting/indentation and I replaced Mailchimp's template tags with Jinja2 ones.

2. Adding a new `MailchimpService.send()` method (instead of the existing `MailchimpService.send_template()`) that uses Jinja2 to render the new template and then calls the Mailchimp API passing the rendered text to send the email.

3. Adding a new Celery task `send()` (instead of the existing `send_template()` task) that calls `MailchimpService.send()` instead of `send_template()`.

4. Changing `DigestService` to call the new `send()` task instead of the old `send_template()` task when sending digest emails.

I also added one image file that was previously hosted on Mailchimp into our app's static files. The new email template references this file.

The original `MailchimpService.send_template()` method and `send_template()` Celery task are still present: these will be needed in case there are any `send_template` messages remaining on the queue. A follow-up commit will remove these methods and their tests once the queue has been drained. (https://github.com/hypothesis/lms/pull/5659)

## Testing

1. Log in to https://hypothesis.instructure.com/ as `eng+canvasteacher@hypothes.is` and launch [localhost (make devdata) HTML Assignment](https://hypothesis.instructure.com/courses/125/assignments/873). This is to make sure that you have a record of the instructor in your DB.

2. Log out and log back in as `eng+canvasstudent@hypothes.is`. Launch the same assignment again and create some annotations.

3. Go to http://localhost:8001/admin/email, enter your email address, change the since and until dates so that they cover the time when you created the annotations, and click send.

4. Log in to <https://login.mailchimp.com/> using the username and password in 1Password then go to <https://mandrillapp.com/login/> and use the **Login using Mailchimp** button. On the Mandrill dashboard hover over the Hypothesis username in the top-right and click **Turn on test mode**. Go to <https://mandrillapp.com/subaccounts/view?id=devdata> and you should see the email that you just sent. Click on **View Content**. You should see something like this:

   ![image](https://github.com/hypothesis/lms/assets/22498/63f1feb3-ab87-44ee-be32-e7a7efb09f84)
